packages/authorization-updater/src/index.ts

```diff
@@ -1,33 +1,126 @@
+/* eslint-disable functional/immutable-data */
+import { v4 as uuidv4 } from "uuid";
 import { runConsumer } from "kafka-iam-auth";
+import { match } from "ts-pattern";
 import { EachMessagePayload } from "kafkajs";
 import {
   messageDecoderSupplier,
   kafkaConsumerConfig,
   logger,
+  getContext,
 } from "pagopa-interop-commons";
+import {
+  Descriptor,
+  EServiceV2,
+  fromEServiceV2,
+  missingMessageDataError,
+  EServiceId,
+} from "pagopa-interop-models";
+import {
+  AuthorizationService,
+  authorizationServiceBuilder,
+} from "./authorization-service.js";
+
+const getDescriptorFromEvent = (msg: {
+  data: {
+    descriptorId: string;
+    eservice?: EServiceV2;
+  };
+}): {
+  eserviceId: EServiceId;
+  descriptor: Descriptor;
+} => {
+  if (!msg.data.eservice) {
+    throw missingMessageDataError("eservice", "EServiceDescriptorPublished");
+  }
+
+  const eservice = fromEServiceV2(msg.data.eservice);
+  const descriptor = eservice.descriptors.find(() => msg.data.descriptorId);
```

Comment:
I don't get this find. Shouldn't it be something like `const descriptor = eservice.descriptors.find((d) => d.id === msg.data.descriptorId);`?